src/AstSemanticChecker.cpp

```diff
@@ -216,13 +216,15 @@ void AstSemanticChecker::checkProgram(ErrorReport& report, const AstProgram& pro
         }
 
         // check types of arguments
-        for (size_t i = 0; i < fun.getArity(); i++) {
-            auto arg = fun.getArg(i);
-            if (fun.acceptsNumbers(i) && !isNumberType(typeAnalysis.getTypes(arg))) {
-                report.addError("Non-numeric argument for functor", arg->getSrcLoc());
-            }
-            if (fun.acceptsSymbols(i) && !isSymbolType(typeAnalysis.getTypes(arg))) {
-                report.addError("Non-symbolic argument for functor", arg->getSrcLoc());
+        if (fun.getFunction() != FunctorOp::ORD) {
```

Comment:
Could you write this as
```
 if (fun.getFunction() == FunctorOp::ORD) {
    return;
}
```
The for loop handling the normal case is then unchanged.

Reply:
Will do.